MbsAPI/fLmd.c

```diff
@@ -61,1186 +56,1266 @@ int clock_gettime(int clockid, struct timespec* tp)
+    if (0 != val) {
+        return (GETLMD__NOLMDFILE);
+    }
+    // use small table
+    if (pLmdControl->iOffsetSize == 4) {
+        pLmdControl->pOffset4 = (uint32_t*)pLmdControl->pOffset8;
+        pLmdControl->pOffset8 = NULL;
+    }
+    return (LMD__SUCCESS);
 }
 //===============================================================
 uint32_t fLmdOffsetWrite(sLmdControl* pLmdControl)
 {
-  int32_t iReturn;
-  char* pbuf;
-  lmdoff_t current;
-  sMbsHeader* pTableHead;
-  pTableHead=(sMbsHeader*)malloc(16);  // header with 8 bytes data for future use.
-  memset(pTableHead,0,16);
-  pTableHead->iWords=(pLmdControl->iElements+1)*pLmdControl->iOffsetSize/2+4;
-  pTableHead->iType=LMD__TYPE_FILE_INDEX_101_2;
-  /*   printf("Table: words:%d type:%08x offbytes:%d\n", */
-  /*     pTableHead->iWords,pTableHead->iType,pLmdControl->iOffsetSize); */
-  iReturn=fgetpos64(pLmdControl->fFile,(fpos64_t*) &current);
-  iReturn=fLmdWriteBuffer(pLmdControl, (char*)pTableHead,16);
-  free(pTableHead);
-  pbuf=(char*)pLmdControl->pOffset4;  // try short table
-  if(pbuf == NULL) { pbuf=(char*)pLmdControl->pOffset8; }
-  iReturn=fLmdWriteBuffer(pLmdControl, pbuf,
-                          (pLmdControl->iElements+1)*pLmdControl->iOffsetSize);
-  if(pLmdControl->pOffset8) {
-    pLmdControl->pMbsFileHeader->iTableOffset = *(pLmdControl->pOffset8+pLmdControl->iElements);
-  }
-  if(pLmdControl->pOffset4) {
-    pLmdControl->pMbsFileHeader->iTableOffset = *(pLmdControl->pOffset4+pLmdControl->iElements);
-  }
-  if(current/4 != pLmdControl->pMbsFileHeader->iTableOffset) {
-    printf("Table offset mismatch: current:%lld calculated:%lld, cur-cal %lld\n",
-           (long long int)(current/4),
-           (long long int)(pLmdControl->pMbsFileHeader->iTableOffset),
-           (long long int)(current/4-pLmdControl->pMbsFileHeader->iTableOffset));
-    return(LMD__FAILURE);
-  }
-  if(iReturn != (pLmdControl->iElements+1)*pLmdControl->iOffsetSize) {
-    printf("Table write error \n");
-    return(LMD__FAILURE);
-  }
-  return(LMD__SUCCESS);
+    int32_t iReturn;
+    char* pbuf;
+    lmdoff_t current;
+    sMbsHeader* pTableHead;
+    pTableHead = (sMbsHeader*)malloc(16);   // header with 8 bytes data for future use.
+    memset(pTableHead, 0, 16);
+    pTableHead->iWords = (pLmdControl->iElements + 1) * pLmdControl->iOffsetSize / 2 + 4;
+    pTableHead->iType = LMD__TYPE_FILE_INDEX_101_2;
+    /*   printf("Table: words:%d type:%08x offbytes:%d\n", */
+    /*     pTableHead->iWords,pTableHead->iType,pLmdControl->iOffsetSize); */
+    iReturn = fgetpos64(pLmdControl->fFile, (fpos64_t*)&current);
+    iReturn = fLmdWriteBuffer(pLmdControl, (char*)pTableHead, 16);
+    free(pTableHead);
+    pbuf = (char*)pLmdControl->pOffset4;   // try short table
+    if (pbuf == NULL) {
+        pbuf = (char*)pLmdControl->pOffset8;
+    }
+    iReturn = fLmdWriteBuffer(pLmdControl, pbuf, (pLmdControl->iElements + 1) * pLmdControl->iOffsetSize);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'iReturn' is reassigned a value before the old one has been used.](https://app.codacy.com/manual/FairRootGroup/FairRoot/pullRequest?prid=5323596)